### ISSUE TITLE
SQLITE_IOERR: disk I/O error on update to 1.23.7

### ISSUE THREAD

**Issue by younessof2m** [louislam/uptime-kuma#4110]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

Hello

I'v upgrade uptime-kuma 1.23.3 to 1.23.4 and when my pod starting I have this errors logs.

I have the same issue in 1.23.4 -> https://github.com/louislam/uptime-kuma/issues/4039

### 👟 Reproduction steps

Upgrade 1.23.3 to 1.23.7 or 1.23.6

### 👀 Expected behavior

Uptime-kuma starting

### 😓 Actual Behavior

Uptime-kuma won't start

### 🐻 Uptime-Kuma Version

1.23.7

### 💻 Operating System and Arch

unknown

### 🌐 Browser

Google Chrome Version 119.0.6045.123 (Build officiel) (64 bits)

### 🐋 Docker Version

unknown

### 🟩 NodeJS Version

18.18.2

### 📝 Relevant log output

```shell
==> Performing startup jobs and maintenance tasks
==> Starting application with user 0 group 0
Welcome to Uptime Kuma
Your Node.js version: 18.18.2
2023-11-27T08:56:23Z [SERVER] INFO: Welcome to Uptime Kuma
2023-11-27T08:56:23Z [SERVER] INFO: Node Env: production
2023-11-27T08:56:23Z [SERVER] INFO: Inside Container: true
2023-11-27T08:56:23Z [SERVER] INFO: Importing Node libraries
2023-11-27T08:56:23Z [SERVER] INFO: Importing 3rd-party libraries
2023-11-27T08:56:26Z [SERVER] INFO: Creating express and socket.io instance
2023-11-27T08:56:26Z [SERVER] INFO: Server Type: HTTP
2023-11-27T08:56:26Z [SERVER] INFO: Importing this project modules
2023-11-27T08:56:26Z [NOTIFICATION] INFO: Prepare Notification Providers
2023-11-27T08:56:26Z [SERVER] INFO: Version: 1.23.7
2023-11-27T08:56:26Z [DB] INFO: Data Dir: ./data/
2023-11-27T08:56:26Z [SERVER] INFO: Connecting to the Database
2023-11-27T08:56:26Z [DB] INFO: SQLite config:
[ { journal_mode: 'wal' } ]
[ { cache_size: -12000 } ]
2023-11-27T08:56:26Z [DB] INFO: SQLite Version: 3.41.1
2023-11-27T08:56:26Z [SERVER] INFO: Connected
2023-11-27T08:56:26Z [DB] INFO: Your database version: 10
2023-11-27T08:56:26Z [DB] INFO: Latest database version: 10
2023-11-27T08:56:26Z [DB] INFO: Database patch not needed
2023-11-27T08:56:26Z [DB] INFO: Database Patch 2.0 Process
2023-11-27T08:56:26Z [DB] INFO: patch-fix-kafka-producer-booleans.sql is not patched
2023-11-27T08:56:26Z [DB] INFO: patch-fix-kafka-producer-booleans.sql is patching
2023-11-27T08:56:26Z [DB] INFO: Closing the database
2023-11-27T08:56:28Z [DB] INFO: SQLite closed
[Error: ALTER TABLE monitor
    DROP COLUMN kafka_producer_allow_auto_topic_creation_old - SQLITE_IOERR: disk I/O error] {
  errno: 10,
  code: 'SQLITE_IOERR'
}
2023-11-27T08:56:28Z [DB] ERROR: Start Uptime-Kuma failed due to issue patching the database
2023-11-27T08:56:28Z [DB] ERROR: Please submit the bug report if you still encounter the problem after restart: https://github.com/louislam/uptime-kuma/issues
```

**Comment by chakflying** [comment]
- IO error would likely be related to your storage. How is you storage configured and where is the data stored in?
- Why is the docker & OS version unknown? How are you running the application?

**Comment by louislam** [comment]
Since we have only received one report during the period and I just re-checked the SQL file, I don't see anything wrong. I assume that there is something wrong with your database.

If you are using non-docker, you can try to bypass it. 

1. Backup `./uptime-kuma/db/patch-fix-kafka-producer-booleans.sql`
2. Open `./uptime-kuma/db/patch-fix-kafka-producer-booleans.sql` with a text editor and make it empty.
3. Start Uptime Kuma
4. Restore the backup file.

**Comment by younessof2m** [comment]
Hello

@chakflying I deploy uptime-kuma on a kubernetes cluster in GCP, the storage is a ssd persistent volume (premium-rwo).

@louislam so at all update the only solution is to bypass sql script ?

**Comment by kokofixcomputers** [comment]
> @louislam so at all update the only solution is to bypass sql script ?

It may or may not be the only solution. Can you check if you can use some terminal commands to write data into the uptime kuma directory? like `mkdir test` which will make a directory? Is the storage full?

**Comment by willthong** [comment]
Similar error for me on first time running of Uptime Kuma:

```
==> Performing startup jobs and maintenance tasks
==> Starting application with user 0 group 0
Welcome to Uptime Kuma
Your Node.js version: 18.19.0
2024-02-08T12:02:39Z [SERVER] INFO: Welcome to Uptime Kuma
2024-02-08T12:02:39Z [SERVER] INFO: Node Env: production
2024-02-08T12:02:39Z [SERVER] INFO: Inside Container: true
2024-02-08T12:02:39Z [SERVER] INFO: Importing Node libraries
2024-02-08T12:02:39Z [SERVER] INFO: Importing 3rd-party libraries
2024-02-08T12:02:40Z [SERVER] INFO: Creating express and socket.io instance
2024-02-08T12:02:40Z [SERVER] INFO: Server Type: HTTP
2024-02-08T12:02:40Z [SERVER] INFO: Importing this project modules
2024-02-08T12:02:41Z [NOTIFICATION] INFO: Prepare Notification Providers
2024-02-08T12:02:41Z [SERVER] INFO: Version: 1.23.11
2024-02-08T12:02:41Z [DB] INFO: Data Dir: ./data/
2024-02-08T12:02:41Z [SERVER] INFO: Connecting to the Database
Trace: [Error: PRAGMA journal_mode = WAL - SQLITE_IOERR: disk I/O error] {
  errno: 10,
  code: 'SQLITE_IOERR'
}
    at process.unexpectedErrorHandler (/app/server/server.js:1899:13)
    at process.emit (node:events:517:28)
    at emit (node:internal/process/promises:149:20)
    at processPromiseRejections (node:internal/process/promises:283:27)
    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)
```

Here's the `docker-compose.yaml`:

```
version: '3.3'

services:
  uptime-kuma:
    image: louislam/uptime-kuma:1
    container_name: uptime-kuma
    volumes:
      - /sharedfolders/AppData/UptimeKuma/uptime-kuma-data:/app/data
    ports:
      - 3001:3001
    restart: always
```

I also tried your suggestion of replacing `patch-fix-kafka-producer-booleans.sql` with a blank file, but I get the same error.

**Comment by chakflying** [comment]
Please be aware that the use of network file storage for the data volume is not supported.

**Comment by CommanderStorm** [comment]
> it was on a RAID array which caused the issue

I am inferring from this statement, that you resolved this issue and does not require/... changes on our side.
Feedback on if any documentation would benefit from changing would be apprciated.

**Comment by willthong** [comment]
I didn't originally raise this issue, so @younessof2m would have to confirm if this fixes the problem for him. I'll submit a PR for amended documentation - thanks for the support 😊

**Comment by CommanderStorm** [comment]
Okay, miscommunications happend.

I thought you were the original author, instead of somebody chiming in with a maybe-unreltated issue.
I don't understand HOW Raid is causing issues. There may be a miriad of problems like your raid controler being broken, storage being full or other issues.
Without a deeper dive, I don't think this is the cause.

**Comment by apio-sys** [comment]
IMHO RAID arrays do work (of course), as long as you can write to them properly. So test first without Kuma Uptime, but just read/write tests with appropriate user your service is running under, check ownership/user rights, check FS, check available space, check available iNodes, etc. All the basic stuff you would do for any app needing an underlying functional storage system. Search on "sqlite error 10 disk i/o error" will probably give you a lot of things to check.

**Comment by willthong** [comment]
Fair enough - but if my RAID was broken I'd expect all the other containers I'm running off the same RAID array to also be broken? Moreover, as soon as I moved the Uptime Kuma directory to a different drive without RAID, the problem isappeared.

**Comment by apio-sys** [comment]
That sounds strange enough indeed if working for the others... But I'm still having a hard time to imagine this is purely an UK issue. Maybe your other containers don't need data dirs where SQLite can write to? IMHO, it is best to run a monitoring system off a bare metal machine (even a small VPS allthough not "technically" bare metal) since I don't like the docker layer in between which can cause all sorts of hickups/bottlenecks whereas you want your monitoring system as clean and stable as possible to make sure your other Services/servers - containerized or not - are avail to your audience. Just my opinion of course.

**Comment by haoxi911** [comment]
@willthong I am not a kuma user, I came from Google while searching SQLITE_IOERR. 

Something you may just need to make a copy of the database file itself, I mean on the same disk. I learned that disk fragments could lead to SQLITE_IOERR, making a copy of the database file will re-allocate data to the same cluster, see [here](https://stackoverflow.com/a/58432646/2537914).

**Comment by CommanderStorm** [comment]
That sounds like a reasonable suggestion.
We will keep the reproduction steps from the post you linked in mind if somebody in the future comes across this issue.

For now, I don't see us able to do anything=> closing the issue
